frontend/web/components/SegmentOverrides.js

```diff
@@ -559,159 +559,178 @@ class TheComponent extends Component {
-                      <a
-                        target='_blank'
-                        href='https://docs.flagsmith.com/basic-features/managing-segments'
-                        rel='noreferrer'
+        <Permission
+          level='project'
+          permission={'MANAGE_SEGMENTS'}
+          id={this.props.projectId}
+        >
+          {({ permission: manageSegments }) => {
+            return (
+              <div className='mt-2 mb-2'>
+                {!this.props.id &&
+                  !this.props.disableCreate &&
+                  !this.props.showCreateSegment &&
+                  !this.props.readOnly && (
+                    <Flex className='text-left'>
+                      <SegmentSelect
+                        disabled={!!isLimitReached}
+                        projectId={this.props.projectId}
+                        data-test='select-segment'
+                        placeholder='Create a Segment Override...'
+                        filter={filter}
+                        value={this.state.selectedSegment}
+                        onChange={(selectedSegment) =>
+                          this.setState({ selectedSegment }, this.addItem)
+                        }
+                      />
+                    </Flex>
+                  )}
+                {!this.props.showCreateSegment &&
+                  !this.props.readOnly &&
+                  !this.props.disableCreate && (
+                    <div className='text-right'>
+                      <Button
+                        className='mt-2'
+                        onClick={() => {
+                          this.setState({ selectedSegment: null })
+                          this.props.setShowCreateSegment(true)
+                        }}
+                        theme='outline'
+                        disabled={
+                          manageSegmentsEnabled
+                            ? !manageSegments || !!isLimitReached
+                            : false
+                        }
                       >
-                        Check the Docs for more details
-                      </a>
-                      .
-                    </InfoMessage>
-                    <SegmentOverrideLimit
-                      id={this.props.environmentId}
-                      maxSegmentOverridesAllowed={ProjectStore.getMaxSegmentOverridesAllowed()}
+                        Create Feature-Specific Segment
+                      </Button>
+                    </div>
+                  )}
+                {this.props.showCreateSegment && !this.state.segmentEditId && (
+                  <div className='create-segment-overrides'>
+                    <CreateSegmentModal
+                      onComplete={(segment) => {
+                        if (this.state.selectedSegment) {
+                          this.props.setShowCreateSegment(false)
+                        } else {
+                          this.props.setShowCreateSegment(false)
```

Comment:
I think you can remove the if-else statement here and stick to just one if statement, since both branches of the if-else statement call the same thing.